x-pack/plugins/encrypted_saved_objects/server/saved_objects/index.ts

```diff
@@ -96,6 +106,38 @@ export function setupSavedObjects({
           )) as T,
         };
       },
+
+      createPointInTimeFinderAsInternalUser: async <T = unknown, A = unknown>(
+        findOptions: SavedObjectsCreatePointInTimeFinderOptions,
+        dependencies?: SavedObjectsCreatePointInTimeFinderDependencies
+      ): Promise<ISavedObjectsPointInTimeFinder<T, A>> => {
+        const [internalRepository, typeRegistry] = await internalRepositoryAndTypeRegistryPromise;
+        const finder = internalRepository.createPointInTimeFinder<T, A>(findOptions, dependencies);
+        const finderAsyncGenerator = finder.find();
+
+        async function* encryptedFinder() {
+          for await (const res of finderAsyncGenerator) {
+            const encryptedSavedObjects = await pMap(res.saved_objects, async (savedObject) => ({
```

Comment:
__suggestion:__ Inside of the `pMap` callback we can check `service.isRegistered(savedObject.type)` and return `savedObject` as is if it's not registered with ESO eliminating a slight overhead of a `service.decryptAttributes`.

Alternatively we could filter out not registered types directly from the `findOptions.type` filter before making the request, but that would make this API more convoluted and inconsistent with the existing `getDecryptedAsInternalUser`.

Reply:
> something like this? 

Yes! Just in case filter includes non-encrypted SO types too.

> Would it make sense to put similar logic for that method as well?

Yeah, it makes sense, and it'd be great if you can update `getDecryptedAsInternalUser` with this logic as well, thanks. I just didn't want to expand the scope of the PR unless you're up for it.

> I think, that wouldn't work for my case, as I need all results from this search query. I this case, I understand, SO with unregistered types would be omitted. I guess, sticking with you previous suggestion would be the best option

I assumed that consumers would likely call this API only with encrypted SO types, but I agree it might not always be the case. Let's stick with the first option :+1:

Reply:
implemented